src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV1Spec.scala

```diff
@@ -1,62 +1,20 @@
-package org.broadinstitute.dsde.workbench.sam.api
+package org.broadinstitute.dsde.workbench.sam
```

Comment:
I am concerned that this spec is changing so drastically. V1 specs are supposed to remain mostly static (additions are ok) because the v1 api should not change. With large changes to the tests I am concerned behavior is also changing.

Do we need all this change?

Reply:
the majority of the change comes from the fact that we're creating new userId instead inside register user endpoint instead of using a userId that's passed in. Previous `TestSamRoutes` won't work well because it's tied to a `UserInfo` object, which requires a `userId`.

The other part of the change comes from @gpolumbo-broad 's suggestion about reducing duplicated testing setup code.